src/backend/distributed/transaction/backend_data.c

```diff
@@ -720,14 +723,40 @@ AssignDistributedTransactionId(void)
 
 	MyBackendData->transactionId.initiatorNodeIdentifier = localGroupId;
 	MyBackendData->transactionId.transactionOriginator = true;
-	MyBackendData->transactionId.transactionNumber =
-		nextTransactionNumber;
+	MyBackendData->transactionId.transactionNumber = nextTransactionNumber;
 	MyBackendData->transactionId.timestamp = currentTimestamp;
+	MyBackendData->citusInitiatedBackend = true;
 
 	SpinLockRelease(&MyBackendData->mutex);
 }
 
 
+/*
+ * MarkCitusInitiatedCoordinatorBackend sets that coordinator backend is
+ * initiated by Citus.
+ */
+void
+MarkCitusInitiatedCoordinatorBackend(void)
+{
+	SpinLockAcquire(&MyBackendData->mutex);
+
+	MyBackendData->citusInitiatedBackend = true;
+
+	SpinLockRelease(&MyBackendData->mutex);
+}
+
+
+/*
+ * UnmarkCitusInitiatedCoordinatorBackendNoLock unset that coordinator backend is
+ * initiated by Citus.
+ */
+static void
+UnmarkCitusInitiatedCoordinatorBackendNoLock(void)
```

Comment:
Minor: I'd be OK to get rid of this function given that we only use it in only one place and it doesn't do anything interesting 😄 .